src/compile/data/stack.ts

```diff
@@ -218,6 +218,7 @@ export class StackNode extends DataFlowNode {
     // Impute
     if (impute && dimensionFieldDef) {
       const dimensionField = dimensionFieldDef ? vgField(dimensionFieldDef, {binSuffix: 'mid'}) : undefined;
+      const dimensionAs = dimensionFieldDef ? vgField(dimensionFieldDef, {binSuffix: 'mid', forAs: true}) : undefined;
```

Comment:
Why should this be different from dimensionField?
especially given than you impute afterward.

Reply:
The key is a field reference in Vega and we have to escape nesting.

Reply:
I revides the code a bit since it's a bit superfluous to have a ternary with `dimensionFieldDef` if it is already required to pass the if.